juju/juju
# Add appdata-source and appdata-sink test charms

# Description of change

These are cloned from the dummy-source and -sink charms but instead use the config token set to populate application relation data so we can see it flow across a relation.

## QA steps

* Deploy appdata-source and appdata-sink and relate them.
```
juju deploy ./acceptancetests/repository/charms/appdata-source -n 3
juju deploy ./acceptancetests/repository/charms/appdata-sink -n 3
juju relate appdata-source appdata-sink
```
* Set the token on the source application.
```
juju config token="some value"
```
* The leader source unit should take the token and store it as application data in the relation (see the log). 
* The sink units should all receive the application token and set it as status, as well as storing it in `/var/run/appdata-sink/token`.

## Documentation changes

None

## Bug reference

None
